apps/src/gamelab/GameLabJr.interpreted.js

```diff
@@ -39,25 +39,64 @@ function initialize(setupHandler) {
 
 // Behaviors
 
-function addBehavior(sprite, behavior, name) {
-  var index = sprite.behavior_keys.indexOf(name);
-  if (index === -1) {
-    sprite.behavior_keys.push(name);
+function addBehavior(sprite, behavior) {
+  if (!sprite || !behavior) {
+    return;
   }
+  behavior = normalizeBehavior(behavior);
 
-  sprite.behaviors[name] = behavior;
+  if (findBehavior(sprite, behavior) !== -1) {
+    return;
+  }
+  sprite.behaviors.push(behavior);
 }
 
-function removeBehavior(sprite, behavior, name) {
-  var index = sprite.behavior_keys.indexOf(name);
-  if (index > -1) {
-    sprite.behavior_keys.splice(index, 1);
+function removeBehavior(sprite, behavior) {
+  if (!sprite || !behavior) {
+    return;
+  }
+  behavior = normalizeBehavior(behavior);
+
+  var index = findBehavior(sprite, behavior);
+  if (index === -1) {
+    return;
   }
+  sprite.behaviors.splice(index, 1);
 }
 
-function hasBehavior(sprite, behavior, name) {
-  var index = sprite.behavior_keys.indexOf(name);
-  return index > -1;
+function normalizeBehavior(behavior) {
+  if (typeof behavior === 'function')  {
+    behavior = {
+      func: behavior,
+      extraArgs: [],
```

Comment:
Does this mean `follow(cat)` will override `follow(dog)`?  What if I want to follow both?